diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4bd07e4..7a22b90 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -506,14 +506,14 @@ set(PCAP_TYPE "" CACHE STRING "Packet capture type")
 #
 option(ENABLE_REMOTE "Enable remote capture" OFF)
 
-if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux" OR CMAKE_SYSTEM_NAME STREQUAL "OHOS")
     option(BUILD_WITH_LIBNL "Build with libnl" ON)
 endif()
 
 #
 # Additional capture modules.
 #
-if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux" OR CMAKE_SYSTEM_NAME STREQUAL "OHOS")
     option(DISABLE_LINUX_USBMON "Disable Linux usbmon USB sniffing support" OFF)
 endif()
 option(DISABLE_BLUETOOTH "Disable Bluetooth sniffing support" OFF)
@@ -545,7 +545,7 @@ else(APPLE)
     option(DISABLE_DBUS "Disable D-Bus sniffing support" OFF)
 endif(APPLE)
 
-if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux" OR CMAKE_SYSTEM_NAME STREQUAL "OHOS")
     option(DISABLE_RDMA "Disable RDMA sniffing support" OFF)
 
     option(DISABLE_DAG "Disable Endace DAG card support" OFF)
@@ -1291,7 +1291,7 @@ cmake_pop_check_state()
 #
 # Same as in Autoconf.
 #
-if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux" OR CMAKE_SYSTEM_NAME STREQUAL "OHOS")
     check_symbol_exists(__GLIBC__ features.h HAVE_GLIBC)
     if(NOT HAVE_GLIBC)
         check_symbol_exists(__UCLIBC__ features.h HAVE_UCLIBC)
@@ -1932,7 +1932,7 @@ else()
         #
         # No.  Try auto-detecting the obvious types first.
         #
-        if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+        if(CMAKE_SYSTEM_NAME STREQUAL "Linux" OR CMAKE_SYSTEM_NAME STREQUAL "OHOS")
             set(PCAP_TYPE linux)
         elseif(CMAKE_SYSTEM_NAME STREQUAL "Haiku")
             set(PCAP_TYPE haiku)
@@ -2358,7 +2358,7 @@ if(NOT WIN32)
 endif()
 
 # Check for hardware timestamp support.
-if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux" OR CMAKE_SYSTEM_NAME STREQUAL "OHOS")
     check_include_file(linux/net_tstamp.h HAVE_LINUX_NET_TSTAMP_H)
 endif()
 
@@ -2369,7 +2369,7 @@ endif()
 #
 # Various Linux-specific mechanisms.
 #
-if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux" OR CMAKE_SYSTEM_NAME STREQUAL "OHOS")
     # Check for usbmon USB sniffing support.
     if(NOT DISABLE_LINUX_USBMON)
         set(PCAP_SUPPORT_LINUX_USBMON TRUE)
@@ -2512,7 +2512,7 @@ endif()
 
 # Check for Bluetooth sniffing support
 if(NOT DISABLE_BLUETOOTH)
-    if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+    if(CMAKE_SYSTEM_NAME STREQUAL "Linux" OR CMAKE_SYSTEM_NAME STREQUAL "OHOS")
         check_include_file(bluetooth/bluetooth.h HAVE_BLUETOOTH_BLUETOOTH_H)
         if(HAVE_BLUETOOTH_BLUETOOTH_H)
             set(PCAP_SUPPORT_BT TRUE)
